### PR TITLE
Add autoinstrumentation for Python

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

**Comment by anuraaga** [comment]
@pavolloffay @jpkrohling

File: pkg/instrumentation/sdk.go [hunk 1]
```diff
@@ -58,6 +58,12 @@ func inject(logger logr.Logger, insts languageInstrumentations, ns corev1.Namesp
 		pod = injectCommonSDKConfig(otelinst, ns, pod)
 		pod = injectNodeJSSDK(logger, otelinst.Spec.NodeJS, pod)
 	}
+	if insts.Python != nil {
+		otelinst := *insts.Python
+		logger.V(1).Info("injecting instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
```
Comment (pavolloffay) [inline pkg/instrumentation/sdk.go:63]: it would be good to include python in the log message

File: pkg/instrumentation/python.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	} else if idx > -1 {
+		if container.Env[idx].ValueFrom != nil {
+			// TODO add to status object or submit it as an event
+			logger.Info("Skipping Python SDK injection, the container defines PYTHONPATH env var value via ValueFrom", "container", container.Name)
+			return pod
+		}
+		container.Env[idx].Value = fmt.Sprintf("%s:%s:%s", pythonPathPrefix, container.Env[idx].Value, pythonPathSuffix)
+	}
+
+	// Set OTEL_TRACES_EXPORTER to HTTP exporter if not set by user because it is what our autoinstrumentation supports.
```
Comment (pavolloffay) [inline pkg/instrumentation/python.go:51]: python does not support OTLP GRPC?

Comment (anuraaga) [reply pkg/instrumentation/python.go:51]: It does but uses a native extension wrapping gRPC-C. This means the extension can only be loaded on the same python version as when pip install was invoked.

HTTP transport is pure python so avoids causing potential version problems.

Comment (anuraaga) [reply pkg/instrumentation/python.go:51]: By the way related comment here

https://github.com/open-telemetry/opentelemetry-operator/pull/524#discussion_r746518160

File: pkg/instrumentation/sdk.go (hunk 1 above)
Comment (anuraaga) [reply pkg/instrumentation/sdk.go]: Thanks, fixed

**Review by pavolloffay — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]
Approving based on @pavolloffay's approval.

**Pull request merged**